stream-chat-android-client/src/main/java/io/getstream/chat/android/client/extensions/MessageExtensions.kt

```diff
@@ -17,8 +17,29 @@
 package io.getstream.chat.android.client.extensions
 
 import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.core.internal.InternalStreamChatApi
+import java.util.Date
 
 public fun Message.enrichWithCid(cid: String): Message = apply {
     replyTo?.enrichWithCid(cid)
     this.cid = cid
 }
+
+/**
+ * Checks if the message mentions the [user].
+ */
+internal fun Message.containsUserMention(user: User): Boolean {
+    return mentionedUsersIds.contains(user.id) || mentionedUsers.any { mentionedUser -> mentionedUser.id == user.id }
+}
+
+@InternalStreamChatApi
+public val NEVER: Date = Date(0)
+
+/**
+ * Check if the message was created after the given [date].
+ */
+@InternalStreamChatApi
+public fun Message.wasCreatedAfter(date: Date?): Boolean {
```

Comment:
Would it be better to just rely on milis here and check that than compare dates?

Something like:

```
return createdAt?.time ?: createdLocallyAt?.time ?: 0L > date.time
```

That way we don't have to allocate a date object in memory that just sits there?

Reply:
hey @hustoj2 thank you for the suggestion, but including `updatedAt` doesn't make sense in the current implementation as `unreadMentionsCount` is connected with `unreadCount`. The message won't be marked as unread if someone edited it so it will introduce even more inconsistency.